cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -817,6 +824,140 @@ def ceph_storage(ceph_admin):
+        'fsid': ceph_admin.fsid(),
+        'auth_supported': ceph_admin.auth(),
+        'use_syslog': "true",
+        'ceph_public_network': '',
+        'ceph_cluster_network': '',
+        'loglevel': 1,
+        'hostname': socket.gethostname(),
+        'admin_key': encoded_key.decode('ascii'),
+        'kubernetes_key': encoded_key.decode('ascii'),
+    }
+
+    # Install the ceph common utilities.
+    apt_install(['ceph-common'], fatal=True)
+
+    etc_ceph_directory = '/etc/ceph'
+    if not os.path.isdir(etc_ceph_directory):
+        os.makedirs(etc_ceph_directory)
+    charm_ceph_conf = os.path.join(etc_ceph_directory, 'ceph.conf')
+    # Render the ceph configuration from the ceph conf template
+    render('ceph.conf', charm_ceph_conf, ceph_context)
+
+    config = hookenv.config()
+    default_storage = config.get('default-storage')
+
+    # Create the ceph yamls from templates in cdk-addons
+    with TemporaryDirectory() as template_path:
+        addon_path = str(_cdk_addons_template_path())
+        render('rbd-secrets.yaml',
+               '{}/rbd-secrets.yaml'.format(template_path), ceph_context,
+               templates_dir=addon_path)
+        render('rbdplugin.yaml',
+               '{}/rbdplugin.yaml'.format(template_path), ceph_context,
+               templates_dir=addon_path)
+        render('csi-provisioner.yaml',
+               '{}/csi-provisioner.yaml'.format(template_path), ceph_context,
+               templates_dir=addon_path)
+        render('csi-attacher.yaml',
+               '{}/csi-attacher.yaml'.format(template_path), ceph_context,
+               templates_dir=addon_path)
+        ceph_context['pool_name'] = 'ext4-pool'
+        ceph_context['fs_type'] = 'ext4'
+        ceph_context['sc_name'] = 'ceph-ext4'
+        if default_storage == 'ceph-ext4':
+            ceph_context['default'] = True
+        else:
+            ceph_context['default'] = False
+        render('rbd-storage-class.yaml',
+               '{}/rbd-storage-class-ext4.yaml'.format(template_path),
+               ceph_context, templates_dir=addon_path)
+        ceph_context['pool_name'] = 'xfs-pool'
+        ceph_context['fs_type'] = 'ext4'
+        ceph_context['sc_name'] = 'ceph-xfs'
+        if default_storage == 'ceph-xfs' or default_storage == 'auto':
+            ceph_context['default'] = True
+        else:
+            ceph_context['default'] = False
+        render('rbd-storage-class.yaml',
+               '{}/rbd-storage-class-xfs.yaml'.format(template_path),
+               ceph_context, templates_dir=addon_path)
+
+        try:
+            # At first glance this is deceptive. The apply stanza will create
+            # if it doesn't exist, otherwise it will update the entry
+            cmd = ['kubectl', 'apply', '-f', template_path]
+            check_call(cmd)
+        except:  # NOQA
+            # the enlistment in kubernetes failed, return and prepare for
+            # re-exec
+            return
```

Comment:
This blanket `except:` with no logging is a bit worrisome. Can you either make it more specific with `except CalledProcessError:` or log the exception with `traceback.print_exc()`?